src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPConnectorApp.scala

```diff
@@ -28,47 +28,81 @@ object ReadingSFTPConnectorApp extends IOApp {
       data: DataFrame = sparkSession.read
         .format("com.springml.spark.sftp") //TODO: Is this library single threaded?
-        .option("host", config.sftp.sftpHost)
-        .option("username", config.sftp.sftpUser)
-        .option("password", config.sftp.sftpPass)
+        .option("host", sftpHost1)
+        .option("username", sftpUser1)
+        .option("password", sftpPass1)
+        .option("header", true)
         .option("fileType", "csv")
+        .option("delimiter", ",")
         .option("inferSchema", "true")
-        .load("/tmp/spark/sample.psv")
-        .repartition(config.spark.partitions)
-
-      // Some other sample operations
-      //_ = sparkSession.catalog.listDatabases().show(truncate = false)
-      //_ = sparkSession.catalog.listTables().show(truncate = false)
-      //_ = sparkSession.sql("show tables").show(truncate = false)
+        .load(sftpPath1)
 
+      //Testing the content of the dataframe, the time in doing the count can be using to measure time in reading.
       _ = data.printSchema()
-      _ = println(s"##############COUNT: ${data.count()}")
+      _ = println(s"### COUNT: ${data.count()}")
       _ = data.show(false)
 
-      // TODO: Other steps to do after the processing.
+      // Creating databases do not work in Dataproc: https://github.com/mozafari/verdictdb/issues/163
+      //_ = if (sparkSession.catalog.databaseExists("sampledb") == false) sparkSession.sqlContext.sql("create database sampledb USING HIVE")
+      //_ = sparkSession.catalog.setCurrentDatabase("sampledb")
+
+      // https://stackoverflow.com/questions/30664008/how-to-save-dataframe-directly-to-hive
+      _ = sparkSession.sqlContext.sql("DROP TABLE IF EXISTS user_data")
+      _ = data.write.mode(SaveMode.Overwrite).format("parquet").saveAsTable("user_data")
 
-      //      Write dataframe as CSV file to FTP server
-      //      df.write().
-      //      format("com.springml.spark.sftp").
-      //      option("host", "SFTP_HOST").
-      //      option("username", "SFTP_USER").
-      //      option("password", "****").
-      //      option("fileType", "csv").
-      //      save("/ftp/files/sample.csv")
+      // Other possible operations when persisting
+      // data.select(df.col("col1"), df.col("col2"), df.col("col3")).write.mode("overwrite").saveAsTable("schemaName.tableName")
+      // data.write.mode(SaveMode.Overwrite).saveAsTable("dbName.tableName")
+
+      // Some other sample operations with databases and tables
+      //_ = sparkSession.catalog.listDatabases().show(truncate = false)
+      _ = sparkSession.catalog.listTables().show(truncate = false)
+      _ = sparkSession.sql("show tables").show(truncate = false)
+
+      // Sample operations to query the Hive database
+      // dataFromHive = sparkSession.sql("select * from sampledb.user_data")
+      // justName = dataFromHive.select("name")
+      dataFromHive = sparkSession.sql("select name from user_data")
```

Comment:
`dataFromHive` is not actually from Hive, right?

Reply:
It is a query done over the user_data persisted table in Hive. so it is a dataframe that comes from there.